aws/sdk/integration-tests/transcribestreaming/tests/test.rs

```diff
@@ -0,0 +1,169 @@
+                        let first_alternative = &result.alternatives.as_ref().unwrap()[0];
+                        full_message += first_alternative.transcript.as_ref().unwrap();
+                        full_message.push('\n');
+                    }
+                }
+            }
+            otherwise => panic!("received unexpected event type: {:?}", otherwise),
+        }
+    }
+
+    // Validate the requests
+    replayer
+        .validate(&["content-type", "content-length"], validate_body)
+        .await
+        .unwrap();
+
+    // Validate the responses
+    assert_eq!(
+        "Good day to you transcribe.\nThis is Polly talking to you from the Rust ST K.\n",
+        full_message
+    );
+}
+
+#[tokio::test]
+async fn test_error() {
+    let input_stream = stream! {
+        let pcm = pcm_data();
+        for chunk in pcm.chunks(CHUNK_SIZE).take(1) {
+            yield Ok(AudioStream::AudioEvent(AudioEvent::builder().audio_chunk(Blob::new(chunk)).build()));
+        }
+    };
+    let (replayer, mut output) =
+        start_request("us-east-1", include_str!("error.json"), input_stream).await;
+
+    match output.transcript_result_stream.recv().await {
+        Err(SdkError::ServiceError {
+            err:
+                StartStreamTranscriptionError {
+                    kind: StartStreamTranscriptionErrorKind::BadRequestException(err),
+                    ..
+                },
+            ..
+        }) => {
+            assert_eq!(
+                Some("A complete signal was sent without the preceding empty frame."),
+                err.message()
+            );
+        }
+        otherwise => panic!("Expected BadRequestException, got: {:?}", otherwise),
+    }
+
+    // Validate the requests
+    replayer
+        .validate(&["content-type", "content-length"], validate_body)
+        .await
+        .unwrap();
+}
+
+async fn start_request(
+    region: &'static str,
+    events_json: &str,
+    input_stream: impl Stream<Item = Result<AudioStream, BoxError>> + Send + Sync + 'static,
+) -> (ReplayingConnection, StartStreamTranscriptionOutput) {
+    let events: Vec<Event> = serde_json::from_str(events_json).unwrap();
+    let replayer = ReplayingConnection::new(events);
+
+    let region = Region::from_static(region);
+    let credentials = Credentials::from_keys("test", "test", None);
+    let config = Config::builder()
+        .region(region)
+        .credentials_provider(credentials)
+        .build();
+    let client = Client::from_conf_conn(config, replayer.clone());
+
+    let output = client
+        .start_stream_transcription()
+        .language_code(LanguageCode::EnGb)
+        .media_sample_rate_hertz(8000)
+        .media_encoding(MediaEncoding::Pcm)
+        .audio_stream(input_stream.into())
+        .send()
+        .await
+        .unwrap();
+
+    (replayer, output)
+}
+
+fn validate_body(expected_body: &[u8], actual_body: &[u8]) -> Result<(), Box<dyn StdError>> {
```

Comment:
should we add an event stream verifier to protocol-test-util?

Also, shouldn't this look for more than one message in the body? I think this only validates the first message?

Reply:
Good catch, it was only validating the first message. Fixed.

As for moving the verifier to a common location, I think that makes sense once we have more than one integration test exercising the event request stream. For now, only the Transcribe test is doing this.